lib/client.js

```diff
@@ -21,10 +23,50 @@ var Client = function(wsdl, endpoint, options) {
   this.wsdl = wsdl;
   this._initializeOptions(options);
   this._initializeServices(endpoint);
+
+  this._signers = [];
   this.httpClient = options.httpClient || new HttpClient(options);
 };
 util.inherits(Client, events.EventEmitter);
 
+/**
+* @callback NodeCallback
+* @param {?*} error
+* @param {String} envelope
+*/
+
+/**
+* @callback ClientEnvelopeSigner
+* @param {String} envelope The envelope to sign
+* @param {ClientEnvelopeSignerOptions} options
+* @param {NodeCallback} callback
+*/
+
+/**
+* @typedef {Object} ClientEnvelopeSignerOptions
+* @property {*} args
+* @property {String} method
+* @property {String} location
+*/
+
+/**
+* Add an envelope signer
+* @param {ClientEnvelopeSigner} signer Signer to sign an envelope
+*/
+Client.prototype.addSigner = function(signer) {
+  this._signers.push(signer);
+};
+
+/**
+* Sign an envelope
+* @param {String} envelope The envelope to sign
+* @param {ClientEnvelopeSignerOptions} options Options to pass to the signer
+* @param {NodeCallback} callback Function to be called once the envelope is signed
+*/
+Client.prototype._sign = function(envelope, options, callback) {
```

Comment:
Yeah I guess, that wouldn't be a problem. Only thought was that you could have multiple. Will change to single